### PR TITLE
impl: start to moving shell into js

### PR DESCRIPTION
**Pull request by SamYuan1990**
It seems that we have to moving shell scripts in JS way for implementation.
as in previous, it still call shell in origin repo for impl, hence we need a totally js implementation. https://github.com/sustainable-computing-io/kepler/actions/runs/3757950683/jobs/6385719004#step:4:10

Signed-off-by: Sam Yuan <yy19902439@126.com>

### CODE REVIEW

File: kind.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    shell.sed('-i',"replicas: 2","replicas: 1",file);
+  });
+  if (shell.exec(`kubectl create -f kube-prometheus/manifests/setup`).code !==0){
+    shell.echo('Error: kubectl create kube-prometheus/manifests/setup');
+    shell.exit(1);
+  }
+
+  shell.ls("kube-prometheus/manifests/prometheusOperator-*").forEach(function(file){
+    let cmd = "kubectl create -f "+file;
+    shell.exec(cmd);
```
[the remark this answers is not in the document]
Comment (SamYuan1990) [reply kind.js:112]: The goal is to use kubectl to create k8s instance base on file name.

File: kind.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  })
+
+  shell.ls("kube-prometheus/manifests/prometheus-*").forEach(function(file){
+    let cmd = "kubectl create -f "+file;
+    shell.exec(cmd);
```
[the remark this answers is not in the document]
Comment (SamYuan1990) [reply kind.js:117]: The goal is to use kubectl to create k8s instance base on file name.

**Comment by rootfs** [comment]
for kind cluster, can we first create a reusable process? We can move `cluster-up` to an external repo and artifact. The GH Action then fetches the artifact and run the script to get the kind cluster.

**Comment by SamYuan1990** [comment]
> for kind cluster, can we first create a reusable process? We can move `cluster-up` to an external repo and artifact. The GH Action then fetches the artifact and run the script to get the kind cluster.

GHA implements in nodejs is reusable across github repos as action.
Moving to an external repo ... for reuse with?

**Comment by SamYuan1990** [comment]
> for kind cluster, can we first create a reusable process? We can move `cluster-up` to an external repo and artifact. The GH Action then fetches the artifact and run the script to get the kind cluster.

and anther thing I am worried about, is that when we get scripts from external repo and artifact, we need to be careful as not override current settings for example, any duplicate files?

File: kind.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+async function fetch_kind() {
+    core.info(`get kind cli`);
+    if (shell.exec('go install sigs.k8s.io/kind@v0.15.0').code !== 0) {
+      shell.echo('Error: kind install failed');
+      shell.exit(1);
+    }
+}
+  
+async function prepare_kind() {
```
[the remark this answers is not in the document]
Comment (rootfs) [reply kind.js:13]: @marceloamaral wdyt?

**Comment by SamYuan1990** [comment]
pre off line disussion with @rootfs, close this pr for now.

**Pull request closed without merging**